app/src/androidTest/java/com/stevesoltys/seedvault/PluginTest.kt

```diff
@@ -35,27 +42,41 @@ import kotlin.random.Random
 class PluginTest : KoinComponent {
 
     private val context = InstrumentationRegistry.getInstrumentation().targetContext
-    private val metadataManager: MetadataManager by inject()
     private val settingsManager: SettingsManager by inject()
     private val mockedSettingsManager: SettingsManager = mockk()
-    private val storage = DocumentsStorage(context, metadataManager, mockedSettingsManager)
-    private val backupPlugin: BackupPlugin = DocumentsProviderBackupPlugin(context, storage)
-    private val restorePlugin: RestorePlugin = DocumentsProviderRestorePlugin(context, storage)
+    private val storage = DocumentsStorage(context, mockedSettingsManager)
+
+    private val kvBackupPlugin: KVBackupPlugin = DocumentsProviderKVBackup(context, storage)
+    private val fullBackupPlugin: FullBackupPlugin = DocumentsProviderFullBackup(context, storage)
+    private val backupPlugin: BackupPlugin = DocumentsProviderBackupPlugin(
+        context,
+        storage,
+        kvBackupPlugin,
+        fullBackupPlugin
+    )
+
+    private val kvRestorePlugin: KVRestorePlugin =
+        DocumentsProviderKVRestorePlugin(context, storage)
+    private val fullRestorePlugin: FullRestorePlugin =
+        DocumentsProviderFullRestorePlugin(context, storage)
+    private val restorePlugin: RestorePlugin =
+        DocumentsProviderRestorePlugin(context, storage, kvRestorePlugin, fullRestorePlugin)
 
     private val token = Random.nextLong()
     private val packageInfo = PackageInfoBuilder.newBuilder().setPackageName("org.example").build()
     private val packageInfo2 = PackageInfoBuilder.newBuilder().setPackageName("net.example").build()
 
     @Before
-    fun setup() {
+    fun setup() = runBlocking {
         every { mockedSettingsManager.getStorage() } returns settingsManager.getStorage()
-        storage.rootBackupDir?.deleteContents()
+        storage.rootBackupDir?.deleteContents(context)
```

Comment:
These `deleteContents()` changes are unrelated, but important as they were not yet using the blocking methods to find all files to delete (see #109).